Source/Dafny/Translators/Substituter.cs

```diff
@@ -0,0 +1,918 @@
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.Dafny
+{
+  /// <summary>
+  /// The substituter has methods to create an expression from an existing one, where the new one has the indicated
+  /// substitutions for "this" (receiverReplacement), variables (substMap), and types (typeMap).
+  /// CAUTION:  The result of the substitution is intended for use by TrExpr, not for well-formedness checks.  In
+  /// particular, the substituter does not copy parts of an expression that are used only for well-formedness checks.
+  /// </summary>
+  public class Substituter
```

Comment:
Any code changes? Can't read the diff since this is a code move. The line numbers differ.

Reply:
You can see the changes if you look at the individual commits.

Reply:
It seems like 3e64d65 is a code move, but if the file changed those changes wouldn't be highlighted.  Since the number of LOC differs I'm asking.

Reply:
Ah, I get it. That commit is purely a code move yes.